extensions/summarize-youtube-video-with-chatgpt/src/getAiSummary.tsx

```diff
@@ -0,0 +1,121 @@
+import { Toast, getPreferenceValues, showToast } from "@raycast/api";
+import { Configuration, OpenAIApi } from "openai";
+import getVideoTranscript from "./getVideoTranscript";
+import { usePromise } from "@raycast/utils";
+import { useState } from "react";
+
+const SUMMARY_MAX_CHARS = 12000;
+
+export default async function getAiSummary(video: string, videoTitle?: string) {
+  const [summary, setSummary] = useState<string | undefined>(undefined);
+
+  const { transcriptLoading, rawTranscript } = getVideoTranscript(video);
+
+  const preferences = getPreferenceValues();
+
+  const configuration = new Configuration({
+    apiKey: preferences.openaiApiToken,
+  });
+  const openai = new OpenAIApi(configuration);
+  let temporarySummary = "";
+  let openAiInstructions = "";
+
+  const { isLoading } = usePromise(
+    async () => {
+      showToast({
+        style: Toast.Style.Animated,
+        title: "💡",
+        message: "Summarizing the video",
+      });
+
+      if (rawTranscript && rawTranscript?.length > SUMMARY_MAX_CHARS) {
+        showToast({
+          style: Toast.Style.Animated,
+          title: "❗",
+          message: "That's a long video, hold on.",
+        });
+
+        const transcriptionSummary = rawTranscript?.split(/(?<=\.)/).reduce(
+          (acc, curr) => {
+            if (acc[acc.length - 1].length + curr.length > SUMMARY_MAX_CHARS) {
+              acc.push(curr);
+            } else {
+              acc[acc.length - 1] += curr;
+            }
+            return acc;
+          },
+          [""]
+        );
```

Comment:
I found an issue with this for a YT where the transcript seemingly had very few `.`s. This resulted in blocks that were too long and produced an error. I added this condition which seems to fix the issue:
```ts
if (curr.length > SUMMARY_MAX_CHARS) {
  const splitTranscription = curr.match(/.{1,12000}/g);
  splitTranscription?.forEach((split) => {
    acc.push(split);
  });
} else if (acc[acc.length - 1]...
```

Reply:
Nice, thank you!